### ISSUE TITLE
background_fetch I cannot receive notfication when app terminated

### ISSUE THREAD

**Issue by hammadpervez** [rafaelsetragni/awesome_notifications#412]
I want to set a notification at time to receive and ending notification time so that I cannot receive notification any longer. But on the next day I want to receive notification as I set duration yesterday.

`
  print("Background Task---------- $time");
if (startTimeStr != null &&
    endTimeStr != null &&
    notificationDuration != null) {
  print(
      "Date : ${startTimeStr} , end: ${endTimeStr}, Duration: $notificationDuration  ");
  final startTime = DateTime.parse(startTimeStr).toUtc();
  final endTime = DateTime.parse(endTimeStr).toUtc();
  if (startTime.isBefore(time)) {
    final convertedStartTime = DateTime(time.year, time.month, time.day,
            startTime.hour, startTime.minute)
        .toUtc();
    final convertedEndTime = DateTime(
            time.year, time.month, time.day, endTime.hour, endTime.minute)
        .toUtc();
    print("Notificating Creating");
    final duration = convertedEndTime.difference(convertedStartTime);
    final scheduleTimes = <DateTime>[];
    for (int i = 0; i < duration.inMinutes; i++) {
      ///if (notificationDuration < duration.inMinutes) {
      final mintuesDuration =
          Duration(minutes: int.parse(notificationDuration) * i);
      final mint = convertedStartTime.add(mintuesDuration);

      scheduleTimes.add(mint.toUtc());
      // }
    }
    print(
        "ConvertedTime: $convertedStartTime and EndTime: $convertedEndTime and Duration $duration and Current Time ${time}");
    AwesomeNotifications()
        .createNotification(
      content: NotificationContent(
        channelKey: 'test_channel',
        title: 'T: $convertedStartTime',
        id: 10,
        body: 'Body: $convertedEndTime and Duration $duration\n $time',
        wakeUpScreen: true,
        criticalAlert: true,
      ),
      schedule: NotificationAndroidCrontab(
          repeats: true,
          preciseAlarm: true,
          allowWhileIdle: true,
          initialDateTime: DateTime.now(),
          expirationDateTime: DateTime.now().add(Duration(minutes: 10)),
          timeZone: AwesomeNotifications.utcTimeZoneIdentifier,
          preciseSchedules: List.generate(
              10,
              (index) =>
                  convertedStartTime.add(Duration(minutes: 1)).toUtc())),
    )
        .then((value) {
      print(
          "==============Notification NotificationAndroidCrontab==============");
    }).catchError((error) {
      print(
          "==============Notification Error : NotificationAndroidCrontab=====$error=========");
    });

    // await AwesomeNotificationService.instance.createScheduleNotification(
    //     NotificationScheduleModel(
    //         startTime: convertedStartTime,
    //         endTime: convertedEndTime,
    //         duration: int.parse(notificationDuration)));
    print("Notification Created");
  } else {
    print("Time Before Else ");
  }
} else {
  print("Background Proccess run");
  print("Background Task -> Else Statement");
}
`

Although, then() method is working but it is not sending notification. I have already initialized Awesome Notifications in my main.dart before runApp()

![image](https://user-images.githubusercontent.com/33895363/152660481-9992f722-71bf-4023-a900-615193e56a22.png)

**Comment by utsav-savani** [comment]
Can you please share how you solved your problem? I am not able to get scheduled notifications when the app is terminated.

**Comment by vicXvac** [comment]
> Can you please share how you solved your problem? I am not able to get scheduled notifications when the app is terminated.

I am facing the same thing, its been 4 days now, this is a serious issue

**Comment by rafaelsetragni** [comment]
First, you need to understand what await does. It works locking the execution when it receives a future object. The method then() and catch() do not returns future, instead it only runs on a future, in parallel.

Then, you need to understand that you must run DartPluginRegistrant.ensureInitialized() to have sure that everything was correctly loaded when you're running while app is killed.

Then, for last, you need to ensure that all background processing is allowed, otherwise the schedules will only work once. And this can be deactivated by the OS through battery save mode or even by the own user. Check it and use a good rationale to convince the user to keep it enabled.

**Comment by hammadpervez** [comment]
> Can you please share how you solved your problem? I am not able to get scheduled notifications when the app is terminated.


put this file proguard-rules.pro in android/app/proguard-rules.pro 
and  paste  the following code in android/app/build_gradle inside buildTypes -> release / debug 

 ```
 minifyEnabled false
            shrinkResources false
            proguardFiles getDefaultProguardFile(
                    'proguard-android-optimize.txt'),
                    'proguard-rules.pro'

```
https://github.com/hammadpervez/progurad-flutter    ,

**Comment by rafaelsetragni** [comment]
Hi! Sorry for taking so long to respond to your issue on GitHub. I was focused on releasing the new 0.7.0 version of `awesome_notifications` and `awesome_notifications_fcm` and i didn't had time enough to do both. But now I can answer all your questions.

So now i'm asking you to recreate this topic using the new issue template. There's a lot of missing informations that i need to understand your problem, and all the instructions are already in the new issue template. Also, remember to check beforehand if your issue was posted by another user.

So I will automatically close all previous issues so far.
Sorry for the inconvenience and i will be waiting for your new issue request.

Thank you so much for your support!
